### PR TITLE
Fixes #28158 - Remove csr flag in hostname-change

### PR DESCRIPTION
**Pull request by chris1984**

### CODE REVIEW

File: packages/katello/katello/hostname-change.rb [hunk 1]
```diff
@@ -255,7 +255,7 @@ def run
         STDOUT.puts "\nChecking custom certificates"
         fail_if_file_not_found([@options[:custom_cert], @options[:custom_key], @options[:custom_cert_req]])
```
Comment (ekohl) [inline packages/katello/katello/hostname-change.rb]: You can probably remove it here as well so it doesn't fail if the file doesn't exist.
```suggestion
         fail_if_file_not_found([@options[:custom_cert], @options[:custom_key]])
```

**Comment by johnpmitsch** [comment]
> I think the whole `custom_cert_req` option doesn't make sense anymore.

+1 to removing the whole option and all the instances in the script where it's used

**Comment by chris1984** [comment]
@johnpmitsch new updated worked correctly with removing all references of `csr` Here is the output from my test run:

```bash
[ INFO 2019-10-31T12:12:04 verbose] Executing hooks in group post
  Success!
  * Katello is running at https://newhost.katello.lan
  * To install an additional Foreman proxy on separate machine continue by running:

      foreman-proxy-certs-generate --foreman-proxy-fqdn "$FOREMAN_PROXY" --certs-tar "/root/$FOREMAN_PROXY-certs.tar"
  The full log is at /var/log/foreman-installer/katello.log
[ INFO 2019-10-31T12:12:06 verbose] All hooks in group post finished
[ INFO 2019-10-31T12:12:06 verbose] Installer finished in 401.427702349 seconds
Restarting puppet services
Redirecting to /bin/systemctl restart puppet.service
**** Hostname change complete! ****
```

**Comment by johnpmitsch** [comment]
@chris1984  is this still an option in the scenario answers file?

https://github.com/theforeman/foreman-packaging/blob/1c1ae693067e2fc747c9f293cd29f35b02a84790/packages/katello/katello/hostname-change.rb#L167

If not, it can be removed too. Seems like we could just check the certificate and key anyways.

**Comment by chris1984** [comment]
> @chris1984 is this still an option in the scenario answers file?
> 
> https://github.com/theforeman/foreman-packaging/blob/1c1ae693067e2fc747c9f293cd29f35b02a84790/packages/katello/katello/hostname-change.rb#L167
> 
> If not, it can be removed too. Seems like we could just check the certificate and key anyways.

Thanks for catching that, ill update again and test

**Comment by chris1984** [comment]
@johnpmitsch updated and retested:

```bash
  Success!
  * Katello is running at https://newhost.katello.lan
  * To install an additional Foreman proxy on separate machine continue by running:

      foreman-proxy-certs-generate --foreman-proxy-fqdn "$FOREMAN_PROXY" --certs-tar "/root/$FOREMAN_PROXY-certs.tar"
  The full log is at /var/log/foreman-installer/katello.log
[ INFO 2019-10-31T19:06:33 verbose] All hooks in group post finished
[ INFO 2019-10-31T19:06:33 verbose] Installer finished in 410.142111066 seconds
Restarting puppet services
Redirecting to /bin/systemctl restart puppet.service
**** Hostname change complete! ****
```

**Comment by chris1984** [comment]
@johnpmitsch tested with custom certs as well:

Installer run with custom certs:

```shell
foreman-installer --scenario katello -v \
                      --certs-server-cert "/root/ownca/dhcp-8-29-17.lab.eng.rdu2.redhat.com/dhcp-8-29-17.lab.eng.rdu2.redhat.com.crt" \
                      --certs-server-key "/root/ownca/dhcp-8-29-17.lab.eng.rdu2.redhat.com/dhcp-8-29-17.lab.eng.rdu2.redhat.com.key" \
                      --certs-server-ca-cert "/root/ownca/cacert.crt"
```			  

katello-change-hostname cert1.katello.lan -u admin -p mB9jvP3eZHTWpFm3 -c /root/ownca/cert1.katello.lan/cert1.katello.lan.crt -k /root/ownca/cert1.katello.lan/cert1.katello.lan.key

Installer run with new certs:

```shell
  Success!
  * Katello is running at https://cert1.katello.lan
  * To install an additional Foreman proxy on separate machine continue by running:

      foreman-proxy-certs-generate --foreman-proxy-fqdn "$FOREMAN_PROXY" --certs-tar "/root/$FOREMAN_PROXY-certs.tar"
  The full log is at /var/log/foreman-installer/katello.log
[ INFO 2019-11-01T12:47:30 verbose] All hooks in group post finished
[ INFO 2019-11-01T12:47:30 verbose] Installer finished in 395.949488721 seconds
Restarting puppet services
Redirecting to /bin/systemctl restart puppet.service
**** Hostname change complete! ****
```

**Comment by ekohl** [comment]
Thanks! Just after merging I realized that this package isn't a nightly so until you bump the release, it won't be released.

**Comment by chris1984** [comment]
@ekohl thanks I will open a PR to bump the release

**Comment by tbrisker** [comment]
Should this (and the release bump?) be pulled in to rpm/1.24 as well?

**Comment by ekohl** [comment]
@tbrisker yes, this is a good one to cherry pick. Mind doing the honors?

**Comment by tbrisker** [comment]
@ekohl - sure, https://github.com/theforeman/foreman-packaging/pull/4391

**Pull request merged**